### PR TITLE
feat: [CDS-47750]: Added AWS ASG Service and Infra type

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Added new `Aws-Asg`  Service type
- Added new `asg` environment type / infra 

https://user-images.githubusercontent.com/106532291/206537720-a191aba9-4e31-4387-aa38-ec6130096237.mov

https://user-images.githubusercontent.com/106532291/206494042-d95792f2-5852-4cba-b536-7872a154c5e5.mov

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/75-cd/components/PipelineSteps/Common/GenericServiceSpec/GenericServiceSpec.module.scss [hunk 1]
```diff
@@ -241,3 +241,13 @@
     padding-right: var(--spacing-3) !important;
   }
 }
+
+.nested-section-card-asg {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/Common/GenericServiceSpec/GenericServiceSpec.module.scss:245]: why are we making a asg change in generic service spec?

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/Common/GenericServiceSpec/GenericServiceSpec.module.scss (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/Common/GenericServiceSpec/GenericServiceSpec.module.scss:245]: Will rename this to be generic one

Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/Common/GenericServiceSpec/GenericServiceSpec.module.scss]: Done

**Review by YounixIO — DISMISSED** [review]

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/services/cd-ng-rq/hooks/useGetServiceAccessListQuery.ts [hunk 2]
```diff
@@ -41,6 +41,7 @@ export interface GetServiceAccessListQueryQueryParams {
     | 'ServerlessAwsLambda'
     | 'Ssh'
     | 'WinRm'
+    | 'Asg'
     | 'TAS'
     | 'Asg'
```
Comment (radujomir) [inline src/services/cd-ng-rq/hooks/useGetServiceAccessListQuery.ts:46]: already provided

**Review by radujomir — CHANGES_REQUESTED** [review]

File: src/services/cd-ng-rq/hooks/useGetServiceAccessListQuery.ts (hunk 2 above)
Comment (vikrant-harness) [reply src/services/cd-ng-rq/hooks/useGetServiceAccessListQuery.ts]: Done

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by radujomir — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**